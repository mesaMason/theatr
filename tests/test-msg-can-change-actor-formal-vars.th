dolphin(int weight, int age):
    print("inside dolphin.")
    print("weight = ")
    print(weight)

    receive:
        swim():
            print("inside swim().")
            print("weight = ")
            print(weight)
        eat():
            print("inside eat().")
            print("weight = ")
            weight = weight + 1
            print(weight)
    drop:
        print("inside drop().")
        print("weight = ")
        weight = weight + 1
        print(weight)
    after:
        return;
        
func main() -> int:
    int weight = 10
    int age = 3
    actor d = new dolphin(weight, age)
    bogus() | d
    eat() | d
    swim() | d
    die() | d
    return 0

