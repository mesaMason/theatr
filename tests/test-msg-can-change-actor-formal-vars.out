inside dolphin.
weight = 
10
inside drop().
weight = 
11
inside eat().
weight = 
12
inside swim().
weight = 
12
